Library/Services/Implementation/NfieldSurveySampleService.cs

```diff
@@ -103,10 +107,45 @@ public Task<int> ResetAsync(string surveyId, string respondentKey)
 
             return Client.PutAsJsonAsync<IEnumerable<SampleFilter>>(uri, filters)
                 .ContinueWith(responseMessageTask => responseMessageTask.Result.Content.ReadAsStringAsync().Result)
-                .ContinueWith(stringResult => JsonConvert.DeserializeObject<SampleResetStatus>(stringResult.Result).ResetCount)
+                .ContinueWith(stringResult => JsonConvert.DeserializeObject<BackgroundActivityStatus>(stringResult.Result).ActivityId)
+                .ContinueWith(activityResult => GetActivityResultAsync(activityResult.Result))
+                .Unwrap()
                 .FlattenExceptions();
         }
 
+        /// <summary>
+        /// Recursive method that polls the activity status until it completes.
+        /// </summary>
```

Comment:
Wouldn't it be better to loop instead of recurse?

Reply:
I don't think it's possible without a synchronous `Wait`, or without increasing the .NET version and using async/await (which I don't think we should do as part of this story).

Reply:
Maybe not possible in this one-liner, but I'm sure it's possible.
If someone uses this to wait for something that takes hours then it'll get a stack overflow pretty quickly. I assume.

Reply:
As discussed in person, it turns out this doesn't actually increase the number of frames on the stack.